gridded_etl_tools/utils/zarr_methods.py

```diff
@@ -1209,3 +1212,172 @@ def are_times_in_expected_order(self, times: tuple[datetime.datetime], expected_
+                              to prevent infinite checks"
+                    )
+                    break
+
+            self.info(
+                f"Values check successfully passed.\
+                      Checking dataset took {datetime.timedelta(seconds=time.perf_counter() - start_checking)}"
+            )
+
+        return True
+
+    def get_prod_update_ds(self) -> xr.Dataset:
+        """
+        Get the prod dataset and filter it to the temporal extent of the latest update
+
+        Returns
+        -------
+        prod_ds
+            The production dataset filtered to only the temporal extent of the latest update
+        """
+        prod_ds = self.store.dataset()
+        update_date_range = slice(
+            datetime.datetime.strptime(prod_ds.attrs["update_date_range"][0], "%Y%m%d%H"),
+            datetime.datetime.strptime(prod_ds.attrs["update_date_range"][1], "%Y%m%d%H"),
+        )
+        time_select = {self.time_dim: update_date_range}
+        return prod_ds.sel(**time_select)
+
+    def get_original_ds(self) -> tuple[xr.Dataset, pathlib.Path]:
+        """
+        Get the original dataset and format it equivalently to the production dataset
+
+        Returns
+        ----------
+        orig_ds
+            The original dataset, unformatted
+        orig_file_path
+            The pathlib.Path to the randomly selected original file
+        """
+        orig_file_path = random.choice(list(self.input_files()))
+        if self.protocol == "file":
+            raw_ds = xr.open_dataset(orig_file_path)
+        # Presumes that use_local_zarr_jsons is enabled. This avoids repeating the DL from S#
+        elif self.protocol == "s3":
+            raw_ds = self.zarr_json_to_dataset(zarr_json_path=orig_file_path)
+        # Reformat the dataset such that it can be selected from equivalently to the prod dataset
+        orig_ds = self.reformat_orig_ds(raw_ds, orig_file_path)
+        return orig_ds, orig_file_path
+
+    def reformat_orig_ds(self, orig_ds: xr.Dataset, orig_file_path: pathlib.Path) -> xr.Dataset:
+        """
+        Open and reformat the original dataset so it can be selected from identically to the production dataset
+        Basically re-run key elements of the transform step
+
+        Parameters
+        ----------
+        orig_ds
+            The original dataset, unformatted
+        orig_file_path
+            A pathlib.Path to the randomly selected original file
+
+        Returns
+        -------
+        orig_ds
+            The original dataset, reformatted similarly to the production dataset
+        """
+        # Drop unused coords and dims
+        for coord in ["crs"]:
+            if coord in orig_ds:
+                orig_ds = orig_ds.drop(coord)
+        # Rename coordinates
+        orig_data_var = [var for var in orig_ds.data_vars][0]
+        rename_coords = {orig_data_var: self.data_var()}
+        for coord in ["lat", "lon"]:
```

Comment:
If I'm understanding correctly, I'm a little concerned about this being hard-coded to only support the literal names lat/lon and latitude/longitude. If an ETL doesn't fit that paradigm it either has to completely rewrite this function to override it or it has to force skip the QC. For example, I have seen "long" in the wild. I think maybe the first half of this function is something to be standardized as standarize_coords or something like that, and then if we need to change the literals we could do it for every set there.

Reply:
That makes sense to me.

An alternative approach would be to call `postprocess_zarr` on every dataset and see what happens. I think at least 95% of this step is managed by that single function. I’m just not completely sure it covers every eventuality. Let me test.